text/0000-light-dom.md

```diff
@@ -0,0 +1,341 @@
+
+```js
+import { LightningElement } from "lwc";
+
+// Example of a shadow DOM component
+class ShadowDOMComponent extends LightningElement {
+  static renderMode = `shadow`; // can be omitted since it's default
+}
+
+// Example of a light DOM component
+class LightDOMComponent extends LightningElement {
+  static renderMode = `light`;
+}
+
+// Default value
+console.log(LightningElement.shadow); // true
+```
+
+The `renderMode` property is looked up by the LWC engine when the component is instantiated to determine how it should render. Changing the value of the `renderMode` static property after the instantiation doesn't influence whether components are rendered in the light DOM or in the shadow DOM.
+
+It also means that developers should be careful not to override the `renderMode` static property value when inheriting from another component. Switching a component mode from shadow DOM to light DOM (and vice-versa) in the child class would certainly break logic in the base class.
+
+```js
+import { LightningElement } from "lwc";
+
+class Base extends LightningElement {}
+
+class Child extends Base {
+  // ⚠️ Changing the shadow static property value in a component class inheriting from a base
+  // will certainly class some unexpected issue.
+  static renderMode = `light`;
+}
+```
+
+### `LightningElement.prototype.template`
+
+When rendering to the shadow DOM, `LightningElement.prototype.template` returns the component associated `ShadowRoot`. When rendering to the light DOM, `LightningElement.prototype.template` value is `null`.
+
+### DOM querying
+
+This proposal doesn't change the way component authors query the light DOM. Component can query their children elements via `LightningElement.prototype.querySelector` and `LightningElement.prototype.querySelectorAll`. It means that turning a shadow DOM component to a light DOM one, all the occurrences of `this.template.querySelector` have to replaced `this.querySelector`.
+
+### Template
+
+Any template associated with a light DOM component should use `lwc:render-mode="light"` to indicate that it is a light DOM template:
+
+```html
+<template lwc:render-mode="light">
+  <!-- light DOM content -->
+</template>
+```
+
+Similarly, shadow DOM components can use `lwc:render-mode="shadow"`, but it's not necessary since it's the default value for that attribute.
+
+Setting the attribute may or may not actually result in compiler differences, but it will be required to allow us the flexibility to have the compiler behave differently for light DOM components versus shadow DOM components.
```

Comment:
Can we list out some concrete use cases for this?

Reply:
@ekashida I don't see any reason why anyone would add `lwc:render-mode="shadow"`, since it's the default. But, just for the sake of consistency, we allow it. The value of the directive can either be `shadow` or `light`, nothing else.

Reply:
@abdulsattar I was referring to the need to have a `render-mode` directive. What use case does it solve? Can we mention those use cases here?

Reply:
Added some text here.